extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/OidcClientImpl.java

```diff
@@ -73,6 +76,12 @@ public OidcClientImpl(WebClient client, String tokenRequestUri, String tokenRevo
         this.clientSecretBasicAuthScheme = OidcCommonUtils.initClientSecretBasicAuth(oidcClientConfig);
         this.jwtBearerAuthentication = oidcClientConfig.credentials().jwt().source() == Source.BEARER;
         this.clientJwtKey = jwtBearerAuthentication ? null : OidcCommonUtils.initClientJwtKey(oidcClientConfig, false);
+        if (jwtBearerAuthentication && oidcClientConfig.credentials().jwt().jwtBearerTokenPath().isPresent()) {
+            this.clientAssertionProvider = new ClientAssertionProvider(vertx,
```

Comment:
IMHO there should be a failure to start if the token can not be loaded

Reply:
1. This allows to recover if the file is not available on startup. I didn't find sufficient evidence that this token will always be available before pods start (I expect secret exists, I don't know whether its value is the token from start), although I expect that is a case. This behavior allows to recover during the getting of tokens if file was added / changed. I only read K8/OCP documentation and they don't say it explicitly (maybe it's given and that's why they don't say it; AFAICT these things are implemented by operators and operators pods can be restarting or whatever, don't know.
2. Later, the situation is identical, if the token is expired and a file with a new token is not found, getting of tokens results in OIDCException. What is a difference.
3. OIDC probably starts before StartupEvent can be observed. Keeping this allows users to prepare this token file in startupevent observer method before HTTP server is opened. Not sure if anyone wants to do that (like make some REST client call to get the token file).

It's easy for me to fail to start if the token cannot be loaded. I don't think it will make this mechanism more reliable. ATM when token cannot be loaded on startup, warning log message describes this, I could change it to error message. Is that enough or are you convinced it should be an exception?

Reply:
> IMHO there should be a failure to start if the token can not be loaded

another thought - dynamic tenants can be only validated when they are created, so for them we would fail during the request, right?

Reply:
Never mind, I'll just fail it and we can wait if someone provides feedback for this later.